### DISCUSSION TITLE
Add facilities to translate ebean exceptions to Spring DataAccessException

### DISCUSSION THREAD

**Discussion by zero11-0203** [ebean-orm/ebean#3258] [Backlog]
Hi all,

For better Spring integration, we could add an ExceptionTranslator to translate ebean exceptions to Spring DataAccessException like [mybatis](https://github.com/mybatis/spring/blob/master/src/main/java/org/mybatis/spring/MyBatisExceptionTranslator.java) and [jooq](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jooq/JooqExceptionTranslator.java)?

**Comment by rbygrave** [comment]
I'm not sure. What is the actual utility value of doing this?

Noting that Ebean has an exception hierarchy much like Springs (inspired by Springs data access exceptions and SQLException mapping in fact and works in a similar way with database platform specific mapping based on SQLState and errorCode obtained from the SQLException). 

The mybatis exception translators are mapping from SQLException and that spring jooq one is as well.  We don't have that case with Ebean which already has that SQLException mapping logic.  So it's a different situation to the one with mybatis and jooq.  

It isn't clear to me exactly why you want to map the ebean exception hierarchy to the spring one.  The mapping of SQLException is a different thing / different case.

**Reply by zero11-0203** [reply]
Hi Rob,

Thanks for your quick reply. 

Our team need to work with multiple persistence frameworks within the same application, and we encapsulate persistence code within Spring Repositories. If ebean provides an implementation for Spring's PersistenceExceptionTranslator, Spring will translate ebean exceptions thrown from repositories to Spring DataAccessException automatically, which will allow us to handle Spring  DataAccessException uniformly in our application layer instead of handling different persistence framework specific exceptions.

**Reply by zero11-0203** [reply]
Maybe [Hibernate](https://github.com/joshlong-attic/spring-framework/blob/master/spring-orm/src/main/java/org/springframework/orm/hibernate3/HibernateExceptionTranslator.java) would be a better example, which also has its own exception hierarchy.

**Comment by rob-bygrave** [comment]
> Maybe [Hibernate](https://github.com/joshlong-attic/spring-framework/blob/master/spring-orm/src/main/java/org/springframework/orm/hibernate3/HibernateExceptionTranslator.java) would be a better example, which also has its own exception hierarchy.

Maybe produce a Proof-Of-Concept with a spring repository using ebean and showing an exception being mapped?
